internal/cmd/beta/server/create/create.go

```diff
@@ -0,0 +1,338 @@
+		KeypairName:                   flags.FlagToStringPointer(p, cmd, keypairNameFlag),
+		Labels:                        flags.FlagToStringToStringPointer(p, cmd, labelFlag),
+		NetworkId:                     flags.FlagToStringPointer(p, cmd, networkIdFlag),
+		NetworkInterfaceIds:           flags.FlagToStringSlicePointer(p, cmd, networkInterfaceIdsFlag),
+		SecurityGroups:                flags.FlagToStringSlicePointer(p, cmd, securityGroupsFlag),
+		ServiceAccountMails:           flags.FlagToStringSlicePointer(p, cmd, serviceAccountEmailsFlag),
+		UserData:                      flags.FlagToStringPointer(p, cmd, userDataFlag),
+		Volumes:                       flags.FlagToStringSlicePointer(p, cmd, volumesFlag),
+	}
+
+	if p.IsVerbosityDebug() {
+		modelStr, err := print.BuildDebugStrFromInputModel(model)
+		if err != nil {
+			p.Debug(print.ErrorLevel, "convert model to string for debugging: %v", err)
+		} else {
+			p.Debug(print.DebugLevel, "parsed input values: %s", modelStr)
+		}
+	}
+
+	return &model, nil
+}
+
+func buildRequest(ctx context.Context, model *inputModel, apiClient *iaas.APIClient) iaas.ApiCreateServerRequest {
+	req := apiClient.CreateServer(ctx, model.ProjectId)
+	var labelsMap *map[string]interface{}
+	if model.Labels != nil && len(*model.Labels) > 0 {
+		// convert map[string]string to map[string]interface{}
+		labelsMap = utils.Ptr(map[string]interface{}{})
+		for k, v := range *model.Labels {
+			(*labelsMap)[k] = v
+		}
+	}
+
+	payload := iaas.CreateServerPayload{
+		Name:             model.Name,
+		MachineType:      model.MachineType,
+		AffinityGroup:    model.AffinityGroup,
+		AvailabilityZone: model.AvailabilityZone,
+
+		ImageId:             model.ImageId,
+		KeypairName:         model.KeypairName,
+		SecurityGroups:      model.SecurityGroups,
+		ServiceAccountMails: model.ServiceAccountMails,
+		UserData:            model.UserData,
+		Volumes:             model.Volumes,
+		Labels:              labelsMap,
+	}
+
+	if model.BootVolumePerformanceClass != nil || model.BootVolumeSize != nil || model.BootVolumeDeleteOnTermination != nil || model.BootVolumeSourceId != nil || model.BootVolumeSourceType != nil {
+		payload.BootVolume = &iaas.CreateServerPayloadBootVolume{
+			PerformanceClass:    model.BootVolumePerformanceClass,
+			Size:                model.BootVolumeSize,
+			DeleteOnTermination: model.BootVolumeDeleteOnTermination,
+			Source: &iaas.BootVolumeSource{
+				Id:   model.BootVolumeSourceId,
+				Type: model.BootVolumeSourceType,
+			},
+		}
+	}
+
+	if model.NetworkInterfaceIds != nil || model.NetworkId != nil {
+		payload.Networking = &iaas.CreateServerPayloadNetworking{}
+
+		if model.NetworkInterfaceIds != nil {
+			payload.Networking.CreateServerNetworkingWithNics = &iaas.CreateServerNetworkingWithNics{
+				NicIds: model.NetworkInterfaceIds,
+			}
+		} else if model.NetworkId != nil {
+			payload.Networking.CreateServerNetworking = &iaas.CreateServerNetworking{
+				NetworkId: model.NetworkId,
+			}
+		}
```

Comment:
Also shouldn't these two be mutually exclusive? The user might not know we will give priority to one over the other